cfn/templates/service.template.yml

```diff
@@ -0,0 +1,1754 @@
+
+    {% if router.scaling.latency_target is defined %}
+      {{- set_default(router.scaling, "statistic", "Average") }}
+      {{- update_element(router.scaling, "disable", false) }}
+    {% endif %}
+
+    {{- set_default(router, "no_load_balancer", False) }}
+    {{- update_element(router, "needs_load_balancer", not(router.no_load_balancer)) }}
+
+    {%- if router.destroy %}
+      {{- dump_debug_object("Destroying routing object", router) }}
+
+      {#- Append a record to cause destroy entries to be created #}
+      {% set x=routers.append({
+        "alias": router.alias | v1_alphanumerize,
+        "destroy": true,
+        "create_load_balancer": false,
+        "create_target_group": false,
+        "uses_load_balancer": false,
+        "scaling": {"disable": true},
+      }) %}
+
+    {%- else %}
+      {{- set_default(router, "public_protocol", router.protocol) }}
+
+      {#- Decide about HTTPS #}
+      {%- if router.public_protocol|upper == "HTTP" and router.certificate is defined %}
+        {{- set_default(router, "public_port", 443) }}
+        {{- update_element(router, "public_protocol", "HTTPS") }}
+
+      {%- else %}
+        {#- Waited to set public_port default until after decision about certificate intentionally #}
+        {{- set_default(router, "public_port", 80) }}
+
+      {%- endif %}
+
+      {#- Make sure port numbers are integers #}
+      {{- update_element(router, "public_port", router.public_port | int) }}
+      {{- update_element(router, "container_port", router.container_port | int) }}
+
+      {#- Based on how above decisions were made, decide on ALB/NLB #}
+      {{- update_element(router, "xlb_type", "application" if router.public_protocol|upper in ['HTTP','HTTPS'] else "network") }}
+
+      {%- if router.xlb_type == "application" and router.perform_health_check %}
+        {{- update_element(router.health_check, "path",
+            router.health_check.path | default(service.health_check_path) | default("/")) }}
+        {{- update_element(router.health_check, "matcher",
+            router.health_check.matcher | default(service.health_check_matcher) | default({"HttpCode": "200-299"})) }}
+        {{- update_element(router.health_check, "interval_seconds",
+            router.health_check.interval_seconds | default(service.health_check_interval_seconds) | default(10)) }}
+        {{- update_element(router.health_check, "healthy_threshold_count",
+            router.health_check.healthy_threshold_count | default(service.healthy_threshold_count) | default(2)) }}
+        {{- update_element(router.health_check, "unhealthy_threshold_count",
+            router.health_check.unhealthy_threshold_count | default(service.unhealthy_threshold_count) | default(6)) }}
+        {{- update_element(router.health_check, "timeout_seconds",
+            router.health_check.timeout_seconds | default(service.health_check_timeout_seconds) | default(5)) }}
+        {{- update_element(router.health_check, "protocol",
```

Comment:
💭 Just to make sure I've got the scope of the changes here. We're adding an additional `protocol` and `port` here.

Reply:
yep!